File: java/sqlbuilder/Insert.java
package com.clickability.cms.dataaccess.sqlbuilder;
import java.util.ArrayList;
import java.util.List;
import java.util.StringJoiner;

import com.clickability.cms.dataaccess.sqlbuilder.Query;
import com.clickability.cms.dataaccess.sqlbuilder.Resource;

public class Insert extends Query {

		private final Resource resource;
		public final List<UpdateAssignment> data;
		
		public Insert(Resource resource) {
			this.data = new ArrayList<UpdateAssignment>();
			this.resource = resource;
		}
		
		public Insert add(UpdateAssignment value) {
			this.data.add(value);
			return this;
		}
		
		public String get() {
			StringJoiner columns = new StringJoiner(",");
			StringJoiner values = new StringJoiner(",");
			for (UpdateAssignment assignment : this.data) {
				columns.add(assignment.column);
				values.add("?");
			}
			String query = String.format("insert into %s (%s) values (%s)", this.resource.get(), columns.toString(), values.toString());
			return query;
		}
				
}

File: queryBuilder/src/main/kotlin/queryBuilder/com/inunison/lib/queryBuilder/Insert.kt
package com.clickability.cms.dataaccess.sqlbuilder

import java.util.ArrayList
import java.util.StringJoiner

import com.clickability.cms.dataaccess.sqlbuilder.Query
import com.clickability.cms.dataaccess.sqlbuilder.Resource

class Insert(private val resource: Resource) : Query() {
    val data: List<UpdateAssignment>

    init {
        this.data = ArrayList<UpdateAssignment>()
    }

    fun add(value: UpdateAssignment): Insert {
        this.data.add(value)
        return this
    }

    fun get(): String {
        val columns = StringJoiner(",")
        val values = StringJoiner(",")
        for (assignment in this.data) {
            columns.add(assignment.column)
            values.add("?")
        }
        return String.format("insert into %s (%s) values (%s)", resource.get(), columns.toString(), values.toString())
    }

}
